Add PasswordField to BUI (#31238)
backstage/backstage: .changeset/eighty-cows-care.md

---
'@backstage/ui': minor
---

> **Breaking** We are adding a new component `PasswordField`. We will be removing support for types `password` and `search` on `TextField`.

Reviewer: ```suggestion
**BREAKING**: Added a new `PasswordField` component. As part of this change, the `password` and `search` types have been removed from `TextField`.
```

Generally we keep changesets in third-person + some minor style tweak
